src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -141,6 +141,15 @@ public static int IndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value
                 return -1;
             }
 
+            if (comparisonType == StringComparison.Ordinal)
```

Comment:
Is the special upfront here worth it? If somebody wants to use the Ordinal compare, they should better use the overload that does not take `StringComparison`.

Reply:
Might be calling it to be sure its Ordinal (if you are used to it from `string` use where the default is `CurrentCulture`)